src/charm.py

```diff
@@ -110,335 +203,226 @@ def set_pod_spec(self, event):
+
+            # execute OpenSSL commands
+            check_call(["openssl", "genrsa", "-out", tmp_dir + "/seldon-cert-gen-ca.key", "2048"])
+            check_call(
+                ["openssl", "genrsa", "-out", tmp_dir + "/seldon-cert-gen-server.key", "2048"]
+            )
+            check_call(
+                [
+                    "openssl",
+                    "req",
+                    "-x509",
+                    "-new",
+                    "-sha256",
+                    "-nodes",
+                    "-days",
+                    "3650",
+                    "-key",
+                    tmp_dir + "/seldon-cert-gen-ca.key",
+                    "-subj",
+                    "/CN=127.0.0.1",
+                    "-out",
+                    tmp_dir + "/seldon-cert-gen-ca.crt",
+                ]
+            )
+            check_call(
+                [
+                    "openssl",
+                    "req",
+                    "-new",
+                    "-sha256",
+                    "-key",
+                    tmp_dir + "/seldon-cert-gen-server.key",
+                    "-out",
+                    tmp_dir + "/seldon-cert-gen-server.csr",
+                    "-config",
+                    tmp_dir + "/seldon-cert-gen-ssl.conf",
+                ]
+            )
+            check_call(
+                [
+                    "openssl",
+                    "x509",
+                    "-req",
+                    "-sha256",
+                    "-in",
+                    tmp_dir + "/seldon-cert-gen-server.csr",
+                    "-CA",
+                    tmp_dir + "/seldon-cert-gen-ca.crt",
+                    "-CAkey",
+                    tmp_dir + "/seldon-cert-gen-ca.key",
+                    "-CAcreateserial",
+                    "-out",
+                    tmp_dir + "/seldon-cert-gen-cert.pem",
+                    "-days",
+                    "365",
+                    "-extensions",
+                    "v3_ext",
+                    "-extfile",
+                    tmp_dir + "/seldon-cert-gen-ssl.conf",
+                ]
+            )
+
+            ret_certs = {
+                "cert": Path(tmp_dir + "/seldon-cert-gen-cert.pem").read_text(),
+                "key": Path(tmp_dir + "/seldon-cert-gen-server.key").read_text(),
+                "ca": Path(tmp_dir + "/seldon-cert-gen-ca.crt").read_text(),
+            }
+
+            # cleanup temporary files
+            check_call(["rm", "-f", tmp_dir + "/seldon-cert-gen-*"])
+
+        return ret_certs
+
+    def main(self, _) -> None:
+        """Perform all required actions the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_leader()
+            self._deploy_k8s_resources()
+            self._update_layer()
+        except ErrorWithStatus as error:
+            self.model.unit.status = error.status
+            return
+
+        self.model.unit.status = ActiveStatus()
```

Comment:
`_deploy_k8s_resources()` tells k8s to apply the resources, but does not wait on their deployment.  This means that when we handle the `install` or `config-changed` hooks, they continue to `self.model.unit.status = ActiveStatus()` even though the resources aren't available. 

I think the missing piece is a check on the resources.  something like:
```python
    def main(self, _) -> None:
        """Perform all required actions the Charm."""
        try:
            self._check_container_connection()
            self._check_leader()
            self._deploy_k8s_resources()
            self._update_layer()
        except ErrorWithStatus as error:
            self.model.unit.status = error.status
            return

        if self._is_seldon_alive() == True:
          self.model.unit.status = ActiveStatus()
        else: 
          # set some sort of status thing here?  Or maybe not, as if we set a status here it'll clobber whatever other status was here previously.  Not sure
```

Reply:
Bug will be created to track this issue: making sure K8S resources are deployed.